src/hooks/user/useProfileByIds.ts

```diff
@@ -0,0 +1,63 @@
+import { useMemo } from "react";
+import { useAsync } from "react-use";
+import { useFirestore } from "reactfire";
+import {
+  collection,
+  documentId,
+  getDocs,
+  query,
+  where,
+} from "firebase/firestore";
+import { chunk } from "lodash";
+
+import { COLLECTION_USERS, FIRESTORE_QUERY_IN_ARRAY_MAX_ITEMS } from "settings";
+
+import { UserId, UserWithId } from "types/id";
+
+import { convertToFirestoreKey } from "utils/id";
+
+type UseProfileByIds = (options: {
+  userIds: UserId[];
+}) => UserWithId[] | undefined;
+
+export const useProfileByIds: UseProfileByIds = ({ userIds }) => {
+  const firestore = useFirestore();
+
+  const usersRequests = useMemo(
+    () =>
+      chunk(userIds, FIRESTORE_QUERY_IN_ARRAY_MAX_ITEMS).map((usersChunk) =>
```

Comment:
👍 

[NOTE] Few more uses like this and we can start thinking of a utility function that abstracts the boilerplate away